source/LoadPanel.cpp

```diff
@@ -44,27 +44,45 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 #include "opengl.h"
 
 #include <algorithm>
+#include <cstdlib>
 #include <stdexcept>
+#include <utility>
 
 using namespace std;
 
 namespace {
+	// Return a pair containing settings to use for time formatting.
+	pair<pair<string, string>, size_t> TimestampFormatString(Preferences::DateFormat fmt)
+	{
+		// pair<string, string>: Linux (1st) and Windows (2nd) format strings
+		// size_t: BUF_SIZE
+		if(fmt == Preferences::DateFormat::YMD)
+			return make_pair(make_pair("%F %T", "%F %T"), 26);
+		if(fmt == Preferences::DateFormat::MDY)
+			return make_pair(make_pair("%-I:%M %p on %b %-d, %Y", "%#I:%M %p on %b %#d, %Y"), 25);
+		if(fmt == Preferences::DateFormat::DMY)
+			return make_pair(make_pair("%-I:%M %p on %-d %b %Y", "%#I:%M %p on %#d %b %Y"), 24);
+
+		// Return YYYY-MM-DD by default.
+		return make_pair(make_pair("%F %T", "%F %T"), 26);
+	}
+
 	// Convert a time_t to a human-readable time and date.
 	string TimestampString(time_t timestamp)
 	{
-		static const size_t BUF_SIZE = 24;
-		char buf[BUF_SIZE];
+		pair<pair<string, string>, size_t> fmt = TimestampFormatString(Preferences::GetDateFormat());
+		char* buf = static_cast<char*>(std::malloc(fmt.second));
 
 #ifdef _WIN32
 		tm date;
 		localtime_s(&date, &timestamp);
-		static const char *FORMAT = "%#I:%M %p on %#d %b %Y";
-		return string(buf, strftime(buf, BUF_SIZE, FORMAT, &date));
+		auto str = string(buf, strftime(buf, fmt.second, fmt.first.first.c_str(), &date));
 #else
 		const tm *date = localtime(&timestamp);
-		static const char *FORMAT = "%-I:%M %p on %-d %b %Y";
-		return string(buf, strftime(buf, BUF_SIZE, FORMAT, date));
+		auto str = string(buf, strftime(buf, fmt.second, fmt.first.first.c_str(), date));
```

Comment:
The comment on TimestampFormatString states that the first member of the inner pair is for Linux and the second is for Windows, but these both use fmt.first.first.

Reply:
3cf53c9482fb23157b9a12f5aa560d79f717cda4